vm/compiler/src/coinbase_puzzle/helpers/partial_solution/mod.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+mod bytes;
+mod serialize;
+mod string;
+
+use super::*;
+use snarkvm_algorithms::crypto_hash::sha256d_to_u64;
+
+/// The partial solution for the coinbase puzzle from a prover.
+#[derive(Copy, Clone, Eq, PartialEq, Hash)]
+pub struct PartialSolution<N: Network> {
+    /// The address of the prover.
+    address: Address<N>,
+    /// The nonce for the solution.
+    nonce: u64,
+    /// The commitment for the solution.
+    commitment: KZGCommitment<N::PairingCurve>,
+}
+
+impl<N: Network> PartialSolution<N> {
+    /// Initializes a new instance of the partial solution.
+    pub const fn new(address: Address<N>, nonce: u64, commitment: KZGCommitment<N::PairingCurve>) -> Self {
+        Self { address, nonce, commitment }
+    }
+
+    /// Returns the address of the prover.
+    pub const fn address(&self) -> &Address<N> {
+        &self.address
+    }
+
+    /// Returns the nonce for the solution.
+    pub const fn nonce(&self) -> u64 {
+        self.nonce
+    }
+
+    /// Returns the commitment for the solution.
+    pub const fn commitment(&self) -> &KZGCommitment<N::PairingCurve> {
+        &self.commitment
```

Comment:
```suggestion
    pub const fn commitment(&self) -> KZGCommitment<N::PairingCurve> {
        self.commitment
```
`KZGCommitment` is `Copy`